Mathlib/Order/Incomparable.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2025 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.Antisymmetrization
+
+/-!
+# Incomparability relation
+
+Two values in a preorder are incomparable whenever `¬ a ≤ b` and `¬ b ≤ a`.
+
+## Main declarations
+
+* `IncompRel`: The incomparability relation. `IncompRel r a b` means that `a` and `b` are related in
+  neither direction by `r`.
+-/
+
+open Function OrderDual
+
+variable {α : Type*} {a b c d : α}
+
+section Relation
+
+variable (r : α → α → Prop)
+
+/-- The incomparability relation `IncompRel r a b` means `¬ r a b` and `¬ r b a`. -/
+def IncompRel : α → α → Prop :=
+  AntisymmRel rᶜ
+
+theorem incompRel_iff {r} : IncompRel r a b ↔ ¬ r a b ∧ ¬ r b a := Iff.rfl
+
+theorem IncompRel.not_le [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ a ≤ b := h.1
+theorem IncompRel.not_ge [LE α] (h : IncompRel (· ≤ ·) a b) : ¬ b ≤ a := h.2
+theorem LE.le.not_incompRel [LE α] (h : a ≤ b) : ¬ IncompRel (· ≤ ·) a b := fun h' ↦ h'.not_le h
+
+@[simp]
+theorem antisymmRel_compl : AntisymmRel rᶜ = IncompRel r :=
+  rfl
+
+@[simp]
+theorem incompRel_compl : IncompRel rᶜ = AntisymmRel r := by
+  simp [IncompRel, AntisymmRel, compl]
+
+theorem incompRel_swap : IncompRel (swap r) = IncompRel r :=
+  antisymmRel_swap _
+
+@[refl]
+theorem incompRel_refl [IsIrrefl α r] (a : α) : IncompRel r a a :=
```

Comment:
Small question: should this be `incompRel_refl` or `IncompRel.refl`? Likewise, should the version with implicit arguments (which I haven't added yet) be `incompRel_rfl` or `IncompRel.rfl`?

I suspect the convention we're using on `AntisymmRel` currently is wrong on both counts, but I want to make sure.

Reply:
`IncompRel.refl`/`IncompRel.rfl` so that we can use it with anonymous dot notation